### PR TITLE
Monitoring documentation:

### PR DESCRIPTION
**Pull request by MalloZup**
# What does this pr?

This pr adress partially #165 issue.

- The monitoring documentation was in the libvirt provider. However we need to make it more generic.
- refactor layout to make it generic.
- also improved the documentation for monitoring

## Note for reviewer:

I think we might need to adress better the case of enabled/disabled monitoring at least for libvirt.

In the azure solution this is not already the case , we need to fix other tech debts before using modules

# rendered here 

https://github.com/MalloZup/ha-sap-terraform-deployments/tree/monitoring-docu

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ ___
 - [google cloud](gcp)
 - [aws](aws)
 
+
+## Major features:
```
Comment (arbulu89) [inline README.md]: Personally, I wouldn't put the `:` character in the headers, it looks strange to me at least.

File: doc/monitoring.md [hunk 2]
```diff
@@ -1,26 +1,42 @@
 # Monitoring deploymenet for SHAP:
```
Comment (arbulu89) [inline doc/monitoring.md]: I wouldn't use the SHAP word. It doesn't mean anything outside of our small world hehe

File: doc/monitoring.md [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
 
-The monitoring module will deploy and install the required tools (grafana, prometheus) to monitor your SHAP stack (SAP HANA, HA cluster, etc).
+# Highlevel description:
 
-The monitoring module will need an extra VM. The packages are the same from Uyuni/Suse manager, so we use the same pkg repository.
+The monitoring feature will deploy and install the required tools (grafana, prometheus) to monitor your SHAP stack (SAP HANA, HA cluster, etc).
```
Comment (arbulu89) [inline doc/monitoring.md]: I would remove the SHAP word

File: doc/monitoring.md [hunk 4]
```diff
@@ -1,26 +1,42 @@
...
-The terraform module follows the same conventions as other modules
+The monitoring feature will need an extra instance, that will host grafana/prometheus server for the dashboard visualisation.
+
+# How to use the monitoring solution:
+
+In order to enable disable the monitoring feature, you need to:
+
+* libvirt: remove or add the `monitoring` module in your main.tf
+
+* azure: for azure cloud we deploy by default the monitoring solution. this need to be refactored and unified later with module similars to libvirt see https://github.com/SUSE/ha-sap-terraform-deployments/issues/107
```
Comment (arbulu89) [inline doc/monitoring.md]: Some typos in this sentence:
-``` This needs to be``` (this in upper case)
-```modules similar to libvirt```

File: doc/monitoring.md [hunk 5]
```diff
@@ -1,26 +1,42 @@
...
 `monitored_hosts` this is a list containing the IP addresses of hosts to be monitored. Under the hood this var tell prometheus the IP where to scrape.
-
-See tfvars.example
 ```
 monitored_hosts = ["192.168.110.X", "192.168.110.Y"]
 ```
 
+* optional:
+
+`monitoring_enabled` default True. This variable will install all different supported exporters to the hosts. ( see  the list of supported exporters)
```
Comment (arbulu89) [inline doc/monitoring.md]: Some extra spaces in the sentence. And the parenthesis is after the `.`

File: doc/monitoring.md [hunk 6]
```diff
@@ -1,26 +1,42 @@
...
 
-If you want to disable monitoring for hosts, use:
-`monitoring_enabled: false`
+Currently supported exporters:
 
+- SAP-HANA database exporter
+- HA Cluster exporter (hawk-apiserver)
 
-# Enable the SAP HANA database exporters
+### SAPA HANA database exporter
```
Comment (arbulu89) [inline doc/monitoring.md]: `SAPA`

File: doc/monitoring.md [hunk 7]
```diff
@@ -1,26 +1,42 @@
...
 
 The SAP HANA database data is exported using the [hanadb_exporter](https://github.com/SUSE/hanadb_exporter) prometheus exporter.
```
Comment (arbulu89) [inline doc/monitoring.md]: `s/exported/exporter/`

File: README.md (hunk 1 above)
Comment (MalloZup) [reply README.md]: ok

File: doc/monitoring.md (hunk 3 above)
Comment (MalloZup) [reply doc/monitoring.md]: done

File: doc/monitoring.md (hunk 4 above)
Comment (MalloZup) [reply doc/monitoring.md]: thx

**Comment by MalloZup** [comment]
thx! I have adressed all points

**Comment by MalloZup** [comment]
Yep is still not perfect but we can increase it later

**Pull request merged**